2.3.0 sdk-jdk7 version crashes on some Android versions

### Description

Here's the error in the console:

```
Caused by: java.util.regex.PatternSyntaxException: Syntax error in regexp pattern near index 5:
^(?<address>\S+):(?<port>\d+)$
     ^
	at java.util.regex.Pattern.compileImpl(Native Method)
	at java.util.regex.Pattern.compile(Pattern.java:411)
	at java.util.regex.Pattern.<init>(Pattern.java:394)
	at java.util.regex.Pattern.compile(Pattern.java:381)
	at com.hedera.hashgraph.sdk.ManagedNodeAddress.<clinit>(ManagedNodeAddress.java:8)
```

Something about this regex, Android doesn't like. Only some versions though. 

Downgrading to 2.2.0 fixes the issue.

### Steps to reproduce

1. build our Android app with 2.3.0 jar as a dependency and run it.

### Additional context

_No response_

### Hedera network

mainnet

### Version

2.3.0

### Operating system

Other

--- a/sdk/src/main/java/com/hedera/hashgraph/sdk/ManagedNodeAddress.java
+++ b/sdk/src/main/java/com/hedera/hashgraph/sdk/ManagedNodeAddress.java
@@ -5,8 +5,8 @@ import java.util.Objects;
 import java.util.regex.Pattern;
 
 class ManagedNodeAddress {
-    private static final Pattern HOST_AND_PORT = Pattern.compile("^(?<address>\\S+):(?<port>\\d+)$");
-    private static final Pattern IN_PROCESS = Pattern.compile("^in-process:(?<name>\\S+)$");
+    private static final Pattern HOST_AND_PORT = Pattern.compile("^(\\S+):(\\d+)$");
+    private static final Pattern IN_PROCESS = Pattern.compile("^in-process:(\\S+)$");
 
     // If address is `in-process:.*` this will contain the right side of the `:`
     @Nullable
@@ -27,12 +27,12 @@ class ManagedNodeAddress {
         var inProcessMatcher = IN_PROCESS.matcher(string);
 
         if (hostAndPortMatcher.matches() && hostAndPortMatcher.groupCount() == 2) {
-            var address = hostAndPortMatcher.group("address");
-            var port = hostAndPortMatcher.group("port");
+            var address = hostAndPortMatcher.group(1);
+            var port = hostAndPortMatcher.group(2);
 
             return new ManagedNodeAddress(null, address, Integer.parseInt(port));
         } else if (inProcessMatcher.matches() && inProcessMatcher.groupCount() == 1) {
-            return new ManagedNodeAddress(inProcessMatcher.group("name"), null, 0);
+            return new ManagedNodeAddress(inProcessMatcher.group(1), null, 0);
         } else {
             throw new IllegalStateException("failed to parse node address");
         }

